core/imageboard/image.php

```diff
@@ -217,6 +217,24 @@ public static function by_random(array $tags = [], int $limit_range = 0): ?Image
             return null;
         }
     }
+	
+	public static function by_random_multi(array $tags=[], int $limit_range=1): array
+    {
+        $max = Image::count_images($tags);
+        if ($max < 1) {
+            return null;
+        }		// From Issue #22 - opened by HungryFeline on May 30, 2011.
+        if ($limit_range > 0 && $max > $limit_range) {
+            $max = $limit_range;
+        }
+        $rand = mt_rand(0, $max-1);
+        $set = Search::find_images($rand, $limit_range, $tags, true);
+        if (count($set) > 0) {
+            return $set;
+        } else {
+            return null;
+        }
```

Comment:
Just `return $set;` seems correct here - if there are no results, then an empty array indicates "no results", where `null` is more like "the concept of results does not exist"